runtime/service_integ_test.go

```diff
@@ -533,3 +533,98 @@ vdf  254:80   0  512B  0 disk`
 			"context cancelled while waiting for container %s to exit, err: %v", containerName, ctx.Err())
 	}
 }
+
+func startAndWaitTask(ctx context.Context, t *testing.T, c containerd.Container) string {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+
+	task, err := c.NewTask(ctx, cio.NewCreator(cio.WithStreams(nil, &stdout, &stderr)))
+	require.NoError(t, err, "failed to create task for container %s", c.ID())
+
+	exitCh, err := task.Wait(ctx)
+	require.NoError(t, err, "failed to wait on task for container %s", c.ID())
+
+	err = task.Start(ctx)
+	require.NoError(t, err, "failed to start task for container %s", c.ID())
+
+	select {
+	case exitStatus := <-exitCh:
+		require.NoError(t, exitStatus.Error(), "failed to retrieve exitStatus")
+		require.Equal(t, uint32(0), exitStatus.ExitCode())
+		require.Equal(t, "", stderr.String())
+	case <-ctx.Done():
+		require.Fail(t, "context cancelled",
+			"context cancelled while waiting for container %s to exit, err: %v", c.ID(), ctx.Err())
+	}
+
+	return stdout.String()
+}
+
+func TestRemoveDir_Isolated(t *testing.T) {
+	internal.RequiresIsolation(t)
+	const vmID = 0
+
+	ctx := namespaces.WithNamespace(context.Background(), "default")
+
+	client, err := containerd.New(containerdSockPath, containerd.WithDefaultRuntime(firecrackerRuntime))
+	require.NoError(t, err, "unable to create client to containerd service at %s, is containerd running?", containerdSockPath)
+	defer client.Close()
+
+	image, err := client.Pull(ctx, debianDockerImage, containerd.WithPullUnpack, containerd.WithPullSnapshotter(naiveSnapshotterName))
+	require.NoError(t, err, "failed to pull image %s, is the the %s snapshotter running?", debianDockerImage, naiveSnapshotterName)
+
+	// TODO once Noah's immutable rootfs change is merged, we can use that as our rootfs for all VMs instead of copying
+	// one per-VM
+	rootfsBytes, err := ioutil.ReadFile(defaultVMRootfsPath)
```

Comment:
nit: rewriting the rootfs file is only required if it's going to be used by multiple VMs concurrently, so I think it can be skipped for this test.

Reply:
I saw that as more like a workaround for the lack of immutable rootfs. We are copying from defaultVMRootfsPath to rootfsPath, to make sure that the rootfs won't be compromised.

Do we already have the "Noah's immutable rootfs change"? If so, we can just use defaultVMRootfsPath and probably make the rootfs read-only.

Reply:
Your concern about making sure the rootfs isn't compromised is totally valid in general, however due to the particulars of this test we don't need to worry about it right now. These integ tests are all currently expected to be running inside isolated docker containers, so in the case that the rootfs was actually modified, the modification would be isolated to the container (thanks to overlayfs).

Noah's immutable rootfs is indeed merged, however our CI system has not yet been updated to use it. That's something we absolutely need to do, however IMO it's also a separate effort from what you're currently doing in this PR, so I think it should be addressed separately.

Reply:
Addressed